myst_nb/cache.py

```diff
@@ -0,0 +1,192 @@
+"""
+Implements integration of jupyter-cache
+"""
+import os
+import nbformat as nbf
+from nbclient import execute
+from pathlib import Path
+
+from sphinx.util import logging
+from sphinx.util.osutil import ensuredir
+
+from jupyter_cache.cache.db import NbCacheRecord
+from jupyter_cache import get_cache
+from jupyter_cache.executors import load_executor
+
+logger = logging.getLogger(__name__)
+
+
+def execution_cache(app, env, added, changed, removed, path_cache=None):
+    """
+    If cacheing is required, stages and executes the added or modified notebooks,
+    and caches then for further use.
+    """
+    jupyter_cache = False
+    nb_list = added.union(
+        changed
+    )  # all the added and changed notebooks should be operated on.
+    if env.config["jupyter_execute_notebooks"]:
+        jupyter_cache = env.config["jupyter_cache"]
+
+    if jupyter_cache is True:
+        path_cache = path_cache or Path(env.outdir).parent.joinpath(".jupyter_cache")
+        app.env.path_cache = (
+            path_cache  # TODO: is there a better way to make it accessible?
+        )
+
+        stage_and_execute(env, nb_list, path_cache)
+
+    return nb_list  # TODO: can also compare timestamps for inputs outputs
+
+
+def stage_and_execute(env, nb_list, path_cache):
+    pk_list = None
```

Comment:
could you add a docstring for any user-facing functions? (or make it hidden)

Reply:
Sure, I will make it hidden